### PR TITLE
Updates and Fixes for e2e

### PR DESCRIPTION
**Pull request by aklef**
# Summary | Résumé

This PR proposes quite a few fixes for e2e tests, better documentation about them, what they are, how to run them, common issues, some help for those, and descriptions of what the files do in general. We now also properly support detox testing the new outbreak exposures and, for that matter, any future app screens.

# Test instructions | Instructions pour tester les modifications

Read the new docs, try to run the tests! Feedback on both the docs and the tests are appreciated

### CODE REVIEW

File: e2e/README.md [hunk 1]
```diff
@@ -1,41 +1,124 @@
...
+
+`yarn pre:test:android`
+
+or
+
+`yarn pre:test:ios`
+
+### Metro server
+
+If you haven't recently run a `yarn pre:test:ios` command, or if you've run it but closed the many   prompts it opened, you might not have metro server running. *Metro must be running* for Detox to talk with React and to interact with the app.
```
Comment (timarney) [inline e2e/README.md:47]:
```
many   prompts
```

Need to remove the extra spaces here.

File: e2e/README.md [hunk 2]
```diff
@@ -1,41 +1,124 @@
...
+or `android.aosp` for android simulator
 
-1. `yarn pre:test:android`
-2. `yarn test:android`
+![Succesful Detox run of exploreDemoMenu.e2e.js](SuccesfulDemoMenuTest.png)
+Took 100s on a 2017 13-inch Macbook Pro.
+
+### Run the entire test suite
+
+This takes many minutes, and binds up most of your computer ressources while at it...
```
Comment (timarney) [inline e2e/README.md:71]: Typo - ressources

Comment (timarney) [inline e2e/README.md:71]: Could say `this may take several minutes`

File: e2e/README.md [hunk 3]
```diff
@@ -1,41 +1,124 @@
...
 
 ## Troubleshooting
 
-Error:
+Sorted by "deeper into the rabbit hole":
+
+### Simulator Errors/Warnings
+
+You can dismiss warnings, and they should go away for the lifetime of the simulator.
+If Errors pop up they will overlap the visual items the simulator is trying to show, and thus block the items below them. This is a pain, you need to bring this up with a Dev so those errors can be remedied or handled differently.
```
Comment (timarney) [inline e2e/README.md:86]: Let's update the wording here.

1) This is an important note - let's add Important: the text here.

Let's update this
> This is a pain, you need to bring this up with a Dev so those errors can be remedied or handled differently.

If the issues persist file an issue on Github to see if the error(s) can be remedied or handled differently.

File: test-gallery/README.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+We can use the test-gallery script to get before/after images. Ex: run once on latest master, and again on feature branch.
+
+This node script will take those two runs (two HTML files) from the test-gallery script, parse them, and merges them into one single continuous file. It displays two version of the app side-by-side, for easy visual difference identification.
+
+Original Andréas K.LeF.
+
+## Future Work
+
+It would be really nice to be able to pass a flag
```
Comment (timarney) [inline test-gallery/README.md:19]: Please note here what the flag would be for.

Comment (aklef) [reply test-gallery/README.md:19]: I accidentally commited this without finishing my idea. Will do

File: e2e/README.md (hunk 3 above)
Comment (aklef) [reply e2e/README.md]: I'm not sure to have understood the first part of what you asked, but I did update that last bit.

File: e2e/README.md (hunk 2 above)
Comment (aklef) [reply e2e/README.md]: Updated in lastest commit

Comment (aklef) [reply e2e/README.md]: Updated in lastest commit

File: e2e/README.md (hunk 1 above)
Comment (aklef) [reply e2e/README.md]: Updated in lastest commit

**Comment by aklef** [comment]
More fixes incoming, please hold on merging..

**Comment by aklef** [comment]
Need #1556 merged before this can go in

**Comment by omartehsin1** [comment]
Ran the detox tests, works fine.

File: e2e/README.md (hunk 3 above)
Comment (aklef) [reply e2e/README.md]: I re-read the doc, I think I understood finally. Please check latest and let me know if it's better!

**Comment by aklef** [comment]
Fixed a merge conflict with master

File: e2e/README.md (hunk 3 above)
Comment (timarney) [reply e2e/README.md]: I would like to remove the "This is a pain" wording.

Would suggest

```
Important:

If Errors pop up they will **overlap the visual items** the simulator is trying to show, and thus **block the items below them**.  If the issues persist file an issue on Github to see if the error(s) can be remedied or handled differently.
```

**Comment by timarney** [comment]
Just one more comment

https://github.com/cds-snc/covid-alert-app/pull/1548#discussion_r626505146

Outside of that looks great.

File: e2e/README.md (hunk 3 above)
Comment (aklef) [reply e2e/README.md]: Done

**Review by timarney — APPROVED** [review]

**Pull request merged**